src/org/zaproxy/zap/extension/params/ExtensionParams.java

```diff
@@ -411,6 +423,9 @@ protected void searchForSelectedParam() {
 						extSearch.search(/*".*" + */item.getName() + "=.*", ExtensionSearch.Type.Header, true, false);
 				} else if (HtmlParameter.Type.header.equals(item.getType())) {
 					extSearch.search(item.getName() + ":.*", ExtensionSearch.Type.Header, true, false);
+				} else if (HtmlParameter.Type.multipart.equals(item.getType())) {
+					extSearch.search("(?i)\\s*content-disposition\\s*:.*\\s+name\\s*\\=?\\s*\\\"?" + item.getName(),

```

Comment:
We should quote the name, it might affect the regex (same applies to other types though).

Reply:
Do you mean `+ "\""+ item.getName() + "\""`?

Reply:
With `Pattern.quote`.

Reply:
Ok so:
`extSearch.search("(?i)\\s*content-disposition\\s*:.*\\s+name\\s*\\=?\\s*\\\"?" + Pattern.quote(item.getName()), ExtensionSearch.Type.Request, true, false);` ?

Shall I go ahead and change the others while I'm at it?

Reply:
Yeah. I think so.

Reply:
Thanks @thc202. I’m tied up in a presentation all afternoon, so I’ll tackle that this evening.

Reply:
OK. Regarding the inline comments, +1 :)